File: library/src/main/java/cn/wandersnail/http/upload/UploadListener.java
package cn.wandersnail.http.upload;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import cn.wandersnail.http.TaskInfo;
import okhttp3.Response;

/**
 * date: 2019/8/23 18:08
 * author: zengfansheng
 */
public interface UploadListener<T> extends UploadProgressListener {
    /**
     * 任务上传状态改变
     */
    void onStateChange(@NonNull TaskInfo.State state, @Nullable Throwable t);

    /**
     * 响应结果
     *
     * @param response          原始响应
     * @param convertedResponse 经过设置的转换器转换后的结果
     */
    void onResponseBodyParse(@NonNull Response response, @Nullable T convertedResponse);
}

File: library/src/main/java/com/snail/network/upload/UploadListener.kt
package com.snail.network.upload

import com.snail.network.TaskInfo
import okhttp3.Response

/**
 * 上传监听
 *
 * date: 2019/5/15 11:01
 * author: zengfansheng
 */
interface UploadListener<T> : UploadProgressListener {
    /**
     * 任务上传状态改变
     */
    fun onStateChange(state: TaskInfo.State, t: Throwable?)

    /**
     * 响应结果解析完成
     *
     * @param response 服务端响应数据
     * @param convertedBody 根据转换器转换过的body
     */
    fun onResponseBodyParse(response: Response, convertedBody: T?)
}